src/test/groovy/edu/ie3/datamodel/utils/validation/NodeValidationUtilsTest.groovy

```diff
@@ -0,0 +1,64 @@
+/*
+ * © 2021. TU Dortmund University,
+ * Institute of Energy Systems, Energy Efficiency and Energy Economics,
+ * Research group Distribution grid planning and operation
+ */
+package edu.ie3.datamodel.utils.validation
+
+import static edu.ie3.util.quantities.PowerSystemUnits.KILOVOLT
+import static edu.ie3.util.quantities.PowerSystemUnits.PU
+
+import edu.ie3.datamodel.exceptions.UnsafeEntityException
+import edu.ie3.datamodel.models.voltagelevels.CommonVoltageLevel
+import edu.ie3.util.interval.RightOpenInterval
+
+import edu.ie3.datamodel.exceptions.InvalidEntityException
+import edu.ie3.datamodel.exceptions.ValidationException
+import edu.ie3.test.common.GridTestData
+import spock.lang.Specification
+import tech.units.indriya.quantity.Quantities
+
+class NodeValidationUtilsTest extends Specification {
+
+	def "Smoke Test: Correct node throws no exception"() {
+		given:
+		def node = GridTestData.nodeA
+
+		when:
+		NodeValidationUtils.check(node)
+
+		then:
+		noExceptionThrown()
+	}
+
+	def "The check method recognizes all potential errors for a node"() {
+		when:
+		NodeValidationUtils.check(invalidNode)
+
+		then:
+		Exception ex = thrown()
+		ex.class == expectedException.class
+		ex.message == expectedException.message
+
+		where:
+		invalidNode                                                            	    || expectedException
+		GridTestData.nodeA.copy().voltLvl(null).build()								|| new ValidationException("Expected a voltage level, but got nothing. :-(")
+		GridTestData.nodeA.copy().voltLvl(new CommonVoltageLevel(
+				"null",
+				null,
+				new HashSet<>(Arrays.asList("null")),
+				new RightOpenInterval<>(
+				Quantities.getQuantity(380d, KILOVOLT), Quantities.getQuantity(560d, KILOVOLT)))).build()																	|| new InvalidEntityException("Node has invalid voltage level", invalidNode)
+		GridTestData.nodeA.copy().voltLvl(new CommonVoltageLevel(
+				"zero volt",
+				Quantities.getQuantity(0d, KILOVOLT),
+				new HashSet<>(Arrays.asList("zero volt")),
+				new RightOpenInterval<>(
+				Quantities.getQuantity(380d, KILOVOLT), Quantities.getQuantity(560d, KILOVOLT)))).build()																	|| new InvalidEntityException("Node has invalid voltage level", invalidNode)
+		//GridTestData.nodeA.copy().vTarget(null).build()								|| new NullPointerException("Target voltage (p.u.) is null")
```

Comment:
Why is this commented out? If it is sensible, please make it work, otherwise just remove it.

Reply:
We said null check is not necessary, probably just forgot to remove this test